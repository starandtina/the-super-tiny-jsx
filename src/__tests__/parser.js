const tokenizer = require('../tokenizer')
const parser = require('../parser')

it('parser: without arguments', () => {
  expect(parser()).toMatchSnapshot()
})

it('parser', () => {
  expect(
    parser(
      tokenizer(`
        html {
          body {
            h1 {
              "Hello World!"
            }
          }
        }
      `),
    ),
  ).toMatchSnapshot()
})
