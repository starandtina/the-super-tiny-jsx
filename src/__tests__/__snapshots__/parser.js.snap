// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parser 1`] = `
Object {
  "props": Object {
    "children": Array [
      Object {
        "props": Object {
          "children": Array [
            Object {
              "props": Object {
                "children": Array [
                  Object {
                    "props": Object {
                      "children": Array [
                        Object {
                          "props": Object {
                            "value": "Hello World!",
                          },
                          "type": "STRING",
                        },
                      ],
                      "value": "h1",
                    },
                    "type": "ELEMENT",
                  },
                ],
                "value": "body",
              },
              "type": "ELEMENT",
            },
          ],
          "value": "html",
        },
        "type": "ELEMENT",
      },
    ],
  },
  "type": "DOCUMENT",
}
`;

exports[`parser: without arguments 1`] = `
Object {
  "props": Object {
    "children": Array [],
  },
  "type": "DOCUMENT",
}
`;
